test/Service/ServiceListenerFactoryTest.php

```diff
@@ -7,42 +7,49 @@
 
 namespace ZendTest\Mvc\Service;
 
+use PHPUnit\Framework\MockObject\MockObject;
 use PHPUnit\Framework\TestCase;
-use ReflectionClass;
-use ReflectionProperty;
 use Zend\Mvc\Service\ServiceListenerFactory;
+use Zend\ServiceManager\Exception\ServiceNotCreatedException;
 use Zend\ServiceManager\ServiceManager;
 
 class ServiceListenerFactoryTest extends TestCase
 {
-    public function setUp()
+    /**
+     * @var ServiceManager|MockObject
+     */
+    private $sm;
+    /**
+     * @var ServiceListenerFactory
+     */
+    private $factory;
+
+    public function setUp() : void
     {
-        $sm = $this->sm = $this->getMockBuilder(ServiceManager::class)
+        $this->sm = $this->getMockBuilder(ServiceManager::class)
                                ->setMethods(['get'])
                                ->getMock();
 
         $this->factory  = new ServiceListenerFactory();
     }
 
-    /**
-     * @expectedException        Zend\ServiceManager\Exception\ServiceNotCreatedException
-     * @expectedExceptionMessage The value of service_listener_options must be an array, string given.
-     */
     public function testInvalidOptionType()
     {
+        $this->expectException(ServiceNotCreatedException::class);
```

Comment:
For me it will be much clearer to put these expectations just before the function which should throw the exception.
In that case it doesn't make a big difference, as the other line here is mocking, but in many case it could be wrong adding it at the very beginning part of the test.

Reply:
This exactly mirrors deprecated annotation behavior.
Whole test suite needs a few improvement passes. Not in this PR/at this time.